src/savecontext.cc

```diff
@@ -31,17 +41,19 @@ DDPROF_NOINLINE static const std::byte *get_stack_end_address() {
 // that both attribute and a suppression are required.
 static DDPROF_NO_SANITIZER_ADDRESS size_t
 save_stack(const std::byte *stack_ptr, ddprof::span<std::byte> buffer) {
-  // Use a thread local variable to cache the stack end address per thread
-  thread_local static const std::byte *stack_end{};
-
+  const std::byte *stack_end = s_stack_end;
   if (unlikely(!stack_end)) {
     // Slow path, takes ~30us
-    stack_end = get_stack_end_address();
+    stack_end = cache_stack_end();
```

Comment:
I would possibly just disable any kind of custom events profiling if we don't have this filled.
Though this code path is unlikely:
- We did not hook into pthread_create correctly ?
- We are not using pthreads ?
I'll look more into this